common/djangoapps/enrollment/views.py

```diff
@@ -451,7 +453,19 @@ def post(self, request):
                     )
                     log.warning(msg)
                     return Response(status=status.HTTP_400_BAD_REQUEST, data={"message": msg})
-                response = api.update_enrollment(username, unicode(course_id), mode=mode, is_active=is_active)
+
+                if mode_changed and active_changed and mode == 'credit' and not not provider_id:
```

Comment:
1) Why `not not provider_id` (two "not"s)?
2) I like the idea of enforcing this constraint, but I'm concerned that this check is too specific.  Would it be possible to generalize this?  I'm imagining something like this:
```
REQUIRED_ATTRIBUTES: {
    "credit": ["provider_id"]
}
# ...
missing_attrs = set(REQUIRED_ATTRIBUTES.get(mode, [])) - enrollment_attributes
if len(missing_attrs) > 0:
    # Send a 400 response with an error message containing the missing attributes.
```